compiler/src/org.graalvm.compiler.lir.aarch64/src/org/graalvm/compiler/lir/aarch64/AArch64StringLatin1InflateOp.java

```diff
@@ -0,0 +1,150 @@
+        masm.jmp(ret);
+
+        masm.bind(simdImpl);
+        emitSIMD(masm, ret);
+
+        masm.bind(ret);
+    }
+
+    private void emitScalar(AArch64MacroAssembler masm) {
+        Register srcAddress = asRegister(temp1);
+        Register destAddress = asRegister(temp2);
+
+        Label loop = new Label();
+
+        try (AArch64MacroAssembler.ScratchRegister scratchReg1 = masm.getScratchRegister(); AArch64MacroAssembler.ScratchRegister scratchReg2 = masm.getScratchRegister()) {
+            Register count = scratchReg1.getRegister();
+            Register currData = scratchReg2.getRegister();
+            masm.mov(64, srcAddress, asRegister(src));
+            masm.mov(64, destAddress, asRegister(dst));
+
+            masm.mov(64, count, asRegister(len));
+
+            masm.align(16);
+            masm.bind(loop);
+            masm.ldr(8, currData, AArch64Address.createImmediateAddress(8, AArch64Address.AddressingMode.IMMEDIATE_POST_INDEXED, srcAddress, 1));
+            masm.str(16, currData, AArch64Address.createImmediateAddress(16, AArch64Address.AddressingMode.IMMEDIATE_POST_INDEXED, destAddress, 2));
+            masm.subs(64, count, count, 1);
+            masm.branchConditionally(AArch64Assembler.ConditionFlag.GE, loop);
+        }
+    }
+
+    private void emitSIMD(AArch64MacroAssembler masm, Label ret) {
+        Register srcChunkAddress = asRegister(temp1);
+        Register destChunkAddress = asRegister(temp2);
+        Register destLowV = asRegister(vectorTemp1);
+        Register destHighV = asRegister(vectorTemp2);
+
+        Label simdLoop = new Label();
+
+        try (AArch64MacroAssembler.ScratchRegister scratchRegister1 = masm.getScratchRegister(); AArch64MacroAssembler.ScratchRegister scratchRegister2 = masm.getScratchRegister()) {
+            Register lastChunkAddress = scratchRegister1.getRegister();
+            Register endOfSrcAddress = scratchRegister2.getRegister();
+
+            masm.mov(64, srcChunkAddress, asRegister(src));
+            masm.mov(64, destChunkAddress, asRegister(dst));
+            masm.add(64, endOfSrcAddress, srcChunkAddress, asRegister(len));
+            masm.sub(64, lastChunkAddress, endOfSrcAddress, chunkByteSize);
+
+            masm.align(16);
+            masm.bind(simdLoop);
+            masm.fldr(128, destLowV, AArch64Address.createImmediateAddress(128, AArch64Address.AddressingMode.IMMEDIATE_POST_INDEXED, srcChunkAddress, chunkByteSize));
+            masm.neon.uxtl2VV(AArch64ASIMDAssembler.ElementSize.Byte, destHighV, destLowV);
+            masm.neon.uxtlVV(AArch64ASIMDAssembler.ElementSize.Byte, destLowV, destLowV);
+            masm.fstp(128, destLowV, destHighV, AArch64Address.createImmediateAddress(128, AArch64Address.AddressingMode.IMMEDIATE_PAIR_POST_INDEXED, destChunkAddress, chunkByteSize * 2));
+            masm.cmp(64, srcChunkAddress, lastChunkAddress);
+            masm.branchConditionally(AArch64Assembler.ConditionFlag.LT, simdLoop);
```

Comment:
When comparing addresses you have to used unsigned comparison (LO)

Reply:
Done